driver-mapping/src/main/java/com/datastax/driver/mapping/MappingManager.java

```diff
@@ -218,57 +219,33 @@ public Session getSession() {
 
     @SuppressWarnings("unchecked")
     private <T> Mapper<T> getMapper(Class<T> klass) {
-        Mapper<T> mapper = (Mapper<T>) mappers.get(klass);
-        if (mapper == null) {
-            synchronized (mappers) {
-                mapper = (Mapper<T>) mappers.get(klass);
-                if (mapper == null) {
-                    EntityMapper<T> entityMapper = AnnotationParser.parseEntity(klass, ReflectionMapper.factory(), this);
-                    mapper = new Mapper<T>(this, klass, entityMapper);
-                    Map<Class<?>, Mapper<?>> newMappers = new HashMap<Class<?>, Mapper<?>>(mappers);
-                    newMappers.put(klass, mapper);
-                    mappers = newMappers;
-                }
-            }
+        if (!mappers.containsKey(klass)) {
+            EntityMapper<T> entityMapper = AnnotationParser.parseEntity(klass, ReflectionMapper.factory(), this);
+            Mapper<T> mapper = new Mapper<T>(this, klass, entityMapper);
+            mappers.putIfAbsent(klass, mapper);
         }
-        return mapper;
+        return (Mapper<T>) mappers.get(klass);
```

Comment:
Yeah, i'll change it.
I'm actually re-thinking that whole PR, these entire parts ([1](https://github.com/datastax/java-driver/blob/41d3d72c23f149b13303074046f72877461a3cd7/driver-mapping/src/main/java/com/datastax/driver/mapping/MappingManager.java#L82), [2](https://github.com/datastax/java-driver/blob/41d3d72c23f149b13303074046f72877461a3cd7/driver-mapping/src/main/java/com/datastax/driver/mapping/MappingManager.java#L96), etc...) may not work.

I think the safest tweak will be to go back to the [original ](https://github.com/datastax/java-driver/blob/3.x/driver-mapping/src/main/java/com/datastax/driver/mapping/MappingManager.java#L37) implementation and add
`final Object lock = new Object()`  for each of those maps and synchronize them when constructing. Less pretty, but less error prone i guess.
What do you think?